github/git_commits.go

```diff
@@ -176,13 +176,14 @@ func createSignatureMessage(commit *createCommit) string {
 		message += fmt.Sprintf("parent %s\n", parent)
 	}
 
-	message += fmt.Sprintf("author %s <%s> %d %s\n", *commit.Author.Name, *commit.Author.Email, commit.Author.Date.Unix(), commit.Author.Date.Format("-0700"))
-	commiter := commit.Committer
-	if commiter == nil {
-		commiter = commit.Author
+	message += fmt.Sprintf("author %s <%s> %d %s\n", commit.Author.GetName(), commit.Author.GetEmail(), commit.Author.GetDate().Unix(), commit.Author.GetDate().Format("-0700"))
+	committer := commit.Committer
+	if committer == nil {
+		committer = commit.Author
 	}
```

Comment:
Ad this point, I still think we need a test if `committer==nil` since both `commit.Committer` and `commit.Author` could be `nil`.

Reply:
I check if `commit.Author` is nil at the top of the function since this function cannot work if that does not exist. Nothing modifies commit object in this function so not sure if we need to test for `nil` again?

https://github.com/google/go-github/pull/1198/files#diff-7da0b51e99c908bfec3fe2f646d4a0fdR154

Reply:
Ah, I forgot about that.  Great, thanks!  You are correct... that is sufficient.  I'm trying to avoid crashes and that settles it. I'm going to add a tweak above, though.